### ISSUE TITLE
[JS] 'apache-arrow' adds a bunch of dev dependencies

### ISSUE THREAD

**Issue by universalmind303** [apache/arrow-js#55]
### Describe the bug, including details regarding any error messages, version, and platform.

if you look at the 'apache-arrow' dependencies, there are a lot of dev dependencies listed as dependencies

```json
  "dependencies": {
    "@swc/helpers": "^0.5.11",
    "@types/command-line-args": "^5.2.3",
    "@types/command-line-usage": "^5.0.4",
    "@types/node": "^20.13.0",
    "command-line-args": "^5.2.1",
    "command-line-usage": "^7.0.1",
    "flatbuffers": "^24.3.25",
    "json-bignum": "^0.0.3",
    "tslib": "^2.6.2"
  },
```

It appears that the only runtime dependency is `flatbuffers`. The rest should be removed or moved to `devDependencies`. 

### Component(s)

JavaScript

**Comment by maorleger** [comment]
Hi folks - I was hoping to bump this issue.

Several of the dependencies introduced here appear to be used only for development, build, or testing workflows, but are currently included as runtime dependencies. Moving those packages to `devDependencies` would reduce the install footprint for downstream consumers and shrink the runtime supply-chain surface area. I know the cli portion complicates this a bit but I think it's worth investing in.

This is particularly relevant today given the increasing frequency of npm ecosystem supply-chain incidents. Even when a dependency is well-maintained, every unnecessary runtime dependency adds additional security review, update, and vulnerability management burden for consumers.

Keeping the production dependency graph as small as possible provides benefits beyond package size:

* Reduced supply-chain risk exposure
* Faster installs
* Fewer transitive vulnerability alerts
* Clearer separation between runtime and development requirements

It would be great to revisit these dependencies and determine which can safely be moved to `devDependencies`.

You mentioned accepting contributions - what would you like to see as an acceptable outcome? I am less concerned with `@types` packages but I wonder how easy it might be to remove command-line-args, command-line-usage, and `@swc/helpers` and can help contribute given a little bit of context or agreement on the end-state

Thanks for your consideration!

Related: https://github.com/apache/arrow/pull/44517, https://github.com/apache/arrow-js/issues/52,

**Comment by kou** [comment]
Could you open a PR as your suggestion to discuss details?

**Comment by trxcllnt** [comment]
IIUC `@swc/helpers` is required because we use swc. The rest of them could be removed if we extracted and published `arrow2csv` as a standalone package that depends on the `@apache-arrow/arrow` library.

**Comment by maorleger** [comment]
Thanks for the response - I went ahead and created #441 as mitigation here. I opted to keep `@types/node` in there as they really don't impact runtime behavior and risks.

Would love some feedback on that! 

Regarding `@swc/helpers`, I came away with a different conclusion about where SWC actually runs in this repo, and I want to make sure I'm not missing something on my end.
 
 The published library and CLI are both compiled by gulp-typescript (TSC), not SWC (I think? see gulp/typescript-task.js). A grep -r '@swc/helpers' targets/ on a freshly built tree turns up zero matches in any .js/.mjs/.d.ts; the only helper imports in the published output come from tslib which remains.
 
 The one place SWC enters the picture is the dev-time bin shebangs that go through `@swc-node/register/esm-register` (e.g. `bin/file-to-stream.ts`, `bin/integration.ts`). That path keeps working with `@swc/helpers` in devDeps - I verified by:
- removing it from dependencies
- deleting node_modules + the lockfile
- reinstalling
- running both the test suite and the dev shebangs end-to-end. 

I believe (but correct me if I'm wrong) that SWC will inline any helpers unless [externalHelpers](https://swc.rs/docs/configuration/compilation#jscexternalhelpers) is set to true.
 
 If you can repro a case where the published artifacts need `@swc/helpers` at runtime I'd love to see it, but otherwise I think this one is safe to keep in devDeps 🙇 

Again, thanks for looking!

**Comment by trxcllnt** [comment]
You're right, I forgot we only need it at runtime for `bin/arrow2csv.js`. If we publish that separately, I think we can move all these to devDependencies.
